src/service/viewer-impl.js

```diff
@@ -419,6 +417,12 @@ export class Viewer {
    * @return {boolean}
    */
   hasCapability(name) {
+    // TODO(lannka, #4457): ideally, we should check if viewer has the capability
+    // of CID storage, rather than if it is embedded.
+    if (name == 'cid') {
+      return isIframed(this.win);
```

Comment:
I honestly don't know much about the multiple doc mode and webview mode. But I guess they don't provide cid?